packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -4375,5 +4386,532 @@ describe("Teal Opcodes", function () {
       );
     });
+
+    describe("ByteSub", () => {
+      const stack = new Stack<StackElem>();
+
+      it("should return correct subtraction of two byte arrays", function () {
+        stack.push(hexToByte('0x01'));
+        stack.push(hexToByte('0x01'));
+        let op = new ByteSub([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), parsing.stringToBytes("")); // Byte ""
+
+        stack.push(hexToByte('0x0200'));
+        stack.push(hexToByte('0x01'));
+        op = new ByteSub([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), hexToByte('0x01FF'));
+
+        // returns are smallest possible
+        stack.push(hexToByte('0x0100'));
+        stack.push(hexToByte('0x01'));
+        op = new ByteSub([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), hexToByte('0xFF'));
+
+        stack.push(hexToByte('0x0ffff1234576'));
+        stack.push(hexToByte('0x1202'));
+        op = new ByteSub([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), new Uint8Array([15, 255, 241, 35, 51, 116]));
+
+        // big num
+        stack.push(hexToByte('0x0123457601234576012345760123457601234576012345760123457601234576'));
+        stack.push(hexToByte('0xffffff01ffffffffffffff01234576012345760123457601234576'));
+        op = new ByteSub([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), new Uint8Array([
+          1, 35, 69, 118, 0, 35, 69, 118, 255,
+          35, 69, 118, 1, 35, 69, 119, 0, 0,
+          0, 0, 0, 0, 0, 0, 0, 0, 0,
+          0, 0, 0, 0, 0
+        ]));
+      });
+
+      it("should panic on underflow", function () {
+        stack.push(hexToByte('0x01'));
+        stack.push(hexToByte('0x02'));
+        const op = new ByteSub([], 0);
+        expectRuntimeError(
+          () => op.execute(stack),
+          RUNTIME_ERRORS.TEAL.UINT64_UNDERFLOW
+        );
+      });
+
+      it("should throw error with ByteSub if stack is below min length",
+        execExpectError(
+          stack,
+          [hexToByte('0x10')],
+          new ByteAdd([], 0),
+          RUNTIME_ERRORS.TEAL.ASSERT_STACK_LENGTH
+        )
+      );
+
+      it("should throw error if ByteSub is used with int",
+        execExpectError(stack, [1n, 2n], new ByteSub([], 0), RUNTIME_ERRORS.TEAL.INVALID_TYPE)
+      );
+    });
+
+    describe("ByteMul", () => {
+      const stack = new Stack<StackElem>();
+
+      it("should return correct multiplication of two byte arrays", function () {
+        stack.push(hexToByte('0x10'));
+        stack.push(hexToByte('0x10'));
+        let op = new ByteMul([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), hexToByte('0x0100'));
+
+        stack.push(hexToByte('0x100000000000'));
+        stack.push(hexToByte('0x00'));
+        op = new ByteMul([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), parsing.stringToBytes(''));
```

Comment:
this was incorrect, we're accepting an empty byte array and not `new Uint8Array([0])`. Reverting.

Reply:
but in test above this we are using `hexToByte('0x0100')`, although same operation is used. what's the difference there ?

Reply:
eg.
```
byte "Algorand"
byte ""
b* 
```
this will output => `byte ""` (which is equivalent to `parsing.stringToBytes('')`/`new Uint8Array([])`.

Adding comment in code